File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
# def solution(arr, queries):
    
#     data = [[] for row in range(len(queries))]
#     result= []
    
#     for order,query in enumerate(queries):
#         print("order: ",order,"query: ", query,"-----------------")
        
#         data[order] = arr[query[0]:query[1]]
#         data[order].sort
        
#         for i in data[order]:
#             if i > query[2]:
#                 result.append(data)
#                 break
#         print("result: ",result)

#         if not result[order]:
#             result.append(-1)
        
            
#     return result



def solution(arr, queries):
    result = []
    for query in queries:
        temp_list = []
        for i in range(query[0], query[1] + 1):
            if arr[i] > query[2]:
                temp_list.append(arr[i])
        try:   
            result.append(min(temp_list))
        except:
            result.append(-1)
    return result

        
            

File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
// import java.util.ArrayList;
// import java.util.Collections;

// class Solution {
//     public int[] solution(int[] arr, int[][] queries) {
//         int[] result = new int[queries.length];

//         for(int j=0;j<queries.length;j++){
//             ArrayList<Integer> temp= new ArrayList<Integer>();
//             ArrayList<Integer> arrTemp= new ArrayList<Integer>();
//             int s = queries[j][0];
//             int e = queries[j][1];
//             int k = queries[j][2];
            
//             //s와 e사이의 i들을 저장 temp에 저장
//             for(int i=0;i<arr.length;i++){
//                 if(arr[i] >= s && arr[i] <= e){
//                     temp.add(arr[i]);
//                 }
//             }
//             for(int i=0;i<temp.size();i++){
//                 arrTemp.add(arr[temp.get(i)]);
//             }
//             Collections.sort(arrTemp);
//             for(int i =0;i<arrTemp.size();i++){
//                 if(arrTemp.get(i) > k){
//                     result[j] = arrTemp.get(i);
//                     break;
//                 }
//                 result[j] = -1;
//             }

//         }
//         return result;
//     }
// }

import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr, int[][] queries){
        int[] answer = new int[queries.length];
        Arrays.fill(answer, -1);
        
        for(int idx =0;idx<queries.length;idx++){
            int[] query = queries[idx];
            int s = query[0], e=query[1], k = query[2];
            for(int i=s; i<=e;i++){
                if(arr[i] > k){
                    answer[idx] = answer[idx] == -1 ? arr[i] : Math.min(arr[i],answer[idx]);
                }
            }
        }
        return answer;
        
    }
}